nvme_gw_server.py

```diff
@@ -196,6 +196,31 @@ def start_spdk(self):
         except Exception as ex:
             self.logger.error(f"Unable to initialize SPDK: \n {ex}")
             raise
+
+        spdk_transports = self.nvme_config.get("spdk", "transports", "tcp")
+
+        for trtype in spdk_transports.split():
+            args = {'trtype' : trtype}
+            name = "transport_" + trtype + "_options"
+            options = self.nvme_config.get("spdk", name, "")
+            for opt in options.split():
+                try:
+                    k, v =opt.split('=')
+                    args[k] = v
+                except ValueError as ex:
+                    self.logger.error(
+                        f"Failed to parse spkd {name} ({options}): \n {ex}"
+                    )
+                    raise
+            try:
+                status = self.spdk_rpc.nvmf.nvmf_create_transport(
```

Comment:
Instead of creating the transport on start why not create it when we create the listener. This way we also don't have to specify all transports available from the beginning but just have the transport type as an argument to the listener?

Reply:
Ok. I think I like this idea.